src/main/scala/tile/FPU.scala

```diff
@@ -764,45 +763,38 @@ class FPU(cfg: FPUParams)(implicit p: Parameters) extends FPUModule()(p) {
   // we don't currently support round-max-magnitude (rm=4)
   io.illegal_rm := io.inst(14,12).isOneOf(5, 6) || io.inst(14,12) === 7 && io.fcsr_rm >= 5
 
-  divSqrt_wdata := 0
-  divSqrt_flags := 0
   if (cfg.divSqrt) {
-    require(fLen == 64)
-    val divSqrt_rm = Reg(Bits())
-    val divSqrt_flags_double = Reg(Bits())
-    val divSqrt_wdata_double = Reg(Bits())
-
-    val divSqrt = Module(new hardfloat.DivSqrtRecF64)
-    divSqrt_inReady := Mux(divSqrt.io.sqrtOp, divSqrt.io.inReady_sqrt, divSqrt.io.inReady_div)
-    val divSqrt_outValid = divSqrt.io.outValid_div || divSqrt.io.outValid_sqrt
-    divSqrt.io.inValid := mem_reg_valid && (mem_ctrl.div || mem_ctrl.sqrt) && !divSqrt_in_flight
-    divSqrt.io.sqrtOp := mem_ctrl.sqrt
-    divSqrt.io.a := fpiu.io.out.bits.in.in1
-    divSqrt.io.b := fpiu.io.out.bits.in.in2
-    divSqrt.io.roundingMode := fpiu.io.out.bits.in.rm
-    divSqrt.io.detectTininess := hardfloat.consts.tininess_afterRounding
-
-    when (divSqrt.io.inValid && divSqrt_inReady) {
-      divSqrt_in_flight := true
-      divSqrt_killed := killm
-      divSqrt_single := mem_ctrl.singleOut
-      divSqrt_waddr := mem_reg_inst(11,7)
-      divSqrt_rm := divSqrt.io.roundingMode
-    }
+    val divSqrt_killed = Reg(Bool())
 
-    when (divSqrt_outValid) {
-      divSqrt_wen := !divSqrt_killed
-      divSqrt_wdata_double := sanitizeNaN(divSqrt.io.out, FType.D)
-      divSqrt_in_flight := false
-      divSqrt_flags_double := divSqrt.io.exceptionFlags
+    makeDivSqrt(FType.S, wb_ctrl.singleOut)
+    fLen match {
+      case 32 =>
+      case 64 => makeDivSqrt(FType.D, !wb_ctrl.singleOut)
```

Comment:
Yeah, for the Q extension.  For now, a match error for anything other than 32/64 is the desired behavior